fernet-aws-secrets-manager-rotator/src/main/java/com/macasaet/fernet/aws/secretsmanager/rotation/AbstractFernetKeyRotator.java

```diff
@@ -0,0 +1,233 @@
+        }
+    }
+
+    protected void conditionallyCreateSecret(final String secretId, final String clientRequestToken) {
+        getSecretsManager().assertCurrentStageExists(secretId);
+        try {
+            getSecretsManager().getSecretVersion(secretId, clientRequestToken);
+            getLogger().warn("createSecret: Successfully retrieved secret for {}. Doing nothing.", secretId);
+        } catch (final ResourceNotFoundException rnfe) {
+            createSecret(secretId, clientRequestToken);
+        }
+    }
+
+    /**
+     * Create a Fernet key secret and store it in AWS Secrets Manager with the stage "AWSPENDING". If there is already an
+     * "AWSPENDING" secret, then do nothing.
+     *
+     * @param secretId the ARN of the secret. e.g. arn:aws:secretsmanager:{region}:{account}:secret:{secret-name}
+     * @param clientRequestToken a unique identifier for this rotation request
+     */
+    protected abstract void createSecret(String secretId, String clientRequestToken);
+
+    /**
+     * Validate the Fernet key secret generated by {@link #createSecret(String, String)}. Throw an exception if there is
+     * a problem with the secret that would make it unusable.
+     *
+     * @param secretId
+     *            the ARN of the secret. e.g. arn:aws:secretsmanager:{region}:{account}:secret:{secret-name}
+     * @param clientRequestToken
+     *            a unique identifier for this rotation request
+     */
+    protected abstract void testSecret(String secretId, String clientRequestToken);
+
+    protected void finishSecret(final String secretId, final String clientRequestToken,
+            final Map<String, List<String>> versions) {
+        String currentVersion = null;
+        for (final String versionId : versions.keySet()) {
+            final List<String> versionStages = versions.get(versionId);
+            if (versionStages.contains(CURRENT.getAwsName())) {
+                if (versionId.equals(clientRequestToken)) {
+                    // The correct version is already marked as current, return
+                    getLogger().warn("finishSecret: Version {} already marked as AWSCURRENT for {}", versionId,
+                            secretId);
+                    return;
+                }
+                currentVersion = versionId;
+                break;
+            }
+        }
+        if (currentVersion == null) {
+            throw new IllegalStateException("No AWSCURRENT secret set for " + secretId + ".");
+        }
+        getSecretsManager().rotateSecret(secretId, clientRequestToken, currentVersion);
+        getLogger().info("finishSecret: Successfully set AWSCURRENT stage to version {} for secret {}.",
+                clientRequestToken, secretId);
+    }
+
+    /**
+     * This seeds the random number generator using KMS if and only it hasn't already been seeded.
+     * 
+     * This requires the permission: <code>kms:GenerateRandom</code>
+     */
+    protected void seed() {
```

Comment:
KMS seeding might be overkill for some. Add an environment variable to disable it.

Reply:
Will address here: https://github.com/l0s/fernet-java8/issues/34